src/mainwindow.cpp

```diff
@@ -92,6 +93,21 @@ MainWindow::MainWindow(QWidget *parent) :
     ui->graphicsViewChipset->scale(2.0, 2.0);
     ui->graphicsViewMap->scale(2.0, 2.0);
 
+
+    //initialising size of qsplitter of general tab
+    QDesktopWidget *desktop = new QDesktopWidget;
```

Comment:
I wonder if you could put this into the `ui` private member (I mean, with the Qt Designer thing). I am not very fond of putting Widget things into the `MainWindow` class because:

- that's not its purpose;
- it violates the [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle).